ovs-service/src/main/java/org/dcsa/ovs/util/pagination/Paginator.java

```diff
@@ -0,0 +1,131 @@
+package org.dcsa.ovs.util.pagination;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.util.VisibleForTesting;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class for doing "cursor" based pagination using spring-data.
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class Paginator {
+  private final ObjectMapper objectMapper;
+
+  /**
+   * Parses the incoming request and returns a Cursor for querying jpa repositories.
+   */
+  public Cursor parseRequest(HttpServletRequest request, CursorDefaults cursorDefaults) {
+    String cursor = request.getParameter("cursor");
+    if (cursor != null) {
+      return cursorFromString(cursor);
+    }
+
+    int pageSize = cursorDefaults.getPageSize();
+    String limit = request.getParameter("limit");
+    if (limit != null) {
+      pageSize = Integer.parseInt(limit);
+    }
+
+    return new Cursor(0, pageSize, cursorDefaults.getSortBy());
+  }
+
+  /**
+   * Sets DCSA pagination headers.
+   *
+   * @param totalPages total pages in a result (see also org.springframework.data.domain.Page)
+   */
+  public void setPageHeaders(
+    HttpServletRequest request,
+    HttpServletResponse response,
+    Cursor currentCursor,
+    int totalPages,
+    String... allowedParameters
+  ) {
+    Set<String> allowedParametersSet = allowedParameters.length == 0 ? null : Arrays.stream(allowedParameters).collect(Collectors.toSet());
```

Comment:
Replacing `null` with `Collections.emptySet()` would make `allowedParametersSet` not nullable (also further down)

Reply:
The other setPageHeaders methods is also public so null is still possible. Would have to check or both null and empty then.